### PR TITLE
Playfair cipher in python - CH20

### PR DESCRIPTION
**Pull request by sharanya02**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes.

### Description

Imlementation of playfair cipher using python for CH20

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue #519

### CODE REVIEW

**Comment by sharanya02** [comment]
@Kajol-Kumari @SKAUL05 @Abhijit2505 @plazzy99 @siddharth25pandey please review it and let me know if I should make any changes.Thank you!

**Review by SKAUL05 — CHANGES_REQUESTED** [review]
Why two files

File: playfair.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
+key=input("Enter the key : ")
```
Comment (vatsalkesarwani12) [inline playfair.py]: delete this file

Comment (sharanya02) [reply playfair.py]: Yes done.

**Comment by sharanya02** [comment]
> Why two files
That was by mistake. I deleted the second file.

**Comment by vatsalkesarwani12** [comment]
edit readme.md

**Comment by sharanya02** [comment]
> edit readme.md

Done.

File: Python/cryptography/playfair.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
+key=input("Enter the key : ")
+key=key.replace(" ", "")
+key=key.upper()
+def matrix(x,y,initial):
+    return [[initial for i in range(x)] for j in range(y)]
+    
+result=list()
+for c in key:
```
Comment (SKAUL05) [inline Python/cryptography/playfair.py]: Use meaningful names instead of alphabets.

File: Python/cryptography/playfair.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        else:
+            result.append(chr(i))
+k=0
+my_matrix=matrix(5,5,0) 
+for i in range(0,5): 
+    for j in range(0,5):
+        my_matrix[i][j]=result[k]
+        k+=1
+
+def locindex(c):
```
Comment (SKAUL05) [inline Python/cryptography/playfair.py]: Use Docstrings to document code.

**Review by SKAUL05 — CHANGES_REQUESTED** [review]

File: Python/cryptography/playfair.py (hunk 3 above)
Comment (sharanya02) [reply Python/cryptography/playfair.py]: Done.

**Comment by sharanya02** [comment]
@plazzy99 @SKAUL05 Can you please review the code again now after the changes?

**Comment by SKAUL05** [comment]
Resolve conflicts

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]
Enter the key : abc

 1.Encryption 
 2.Decryption: 
 3.Exit 
 Enter Your Choice: 1
ENTER MSG : hello
CIPHER TEXT: KC NV MP 
 1.Encryption 
 2.Decryption: 
 3.Exit 
 Enter Your Choice: 2
ENTER CIPHER TEXT:hello
PLAIN TEXT: KC FF 
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-1-a0792602bbe2> in <module>()
    107         encrypt()
    108     elif ch==2:
--> 109         decrypt()
    110     elif ch==3:
    111         break;

<ipython-input-1-a0792602bbe2> in decrypt()
     92         loc=locindex(msg[i])
     93         loc1=list()
---> 94         loc1=locindex(msg[i+1])
     95         if loc[1]==loc1[1]:
     96             print("{}{}".format(my_matrix[(loc[0]-1)%5][loc[1]],my_matrix[(loc1[0]-1)%5][loc1[1]]),end=' ')

IndexError: string index out of range


make changes so that if cipher text is not correct code should not crash instead show an error or warning message

**Comment by atarax665** [comment]
@sharanya02 Do the suggested changes and resolve the conflict to get this PR merged ASAP.

**Comment by sharanya02** [comment]
> @sharanya02 Do the suggested changes and resolve the conflict to get this PR merged ASAP.

Done.Can you review it? Thank you!

**Comment by sharanya02** [comment]
> Enter the key : abc
> 
> ## 1.Encryption
> 2.Decryption:
> 3.Exit
> Enter Your Choice: 1
> ENTER MSG : hello
> CIPHER TEXT: KC NV MP
> 1.Encryption
> 2.Decryption:
> 3.Exit
> Enter Your Choice: 2
> ENTER CIPHER TEXT:hello
> PLAIN TEXT: KC FF
> IndexError Traceback (most recent call last)
> in ()
> 107 encrypt()
> 108 elif ch==2:
> --> 109 decrypt()
> 110 elif ch==3:
> 111 break;
> 
> in decrypt()
> 92 loc=locindex(msg[i])
> 93 loc1=list()
> ---> 94 loc1=locindex(msg[i+1])
> 95 if loc[1]==loc1[1]:
> 96 print("{}{}".format(my_matrix[(loc[0]-1)%5][loc[1]],my_matrix[(loc1[0]-1)%5][loc1[1]]),end=' ')
> 
> IndexError: string index out of range
> 
> make changes so that if cipher text is not correct code should not crash instead show an error or warning message

Done.

File: Python/README.md [hunk 4]
```diff
@@ -1,293 +1,3 @@
-
-<p align="center">
-    <img src="../img/neo_algo.png"><br>
```
Comment (vatsalkesarwani12) [inline Python/README.md:3]: why made so many changes .

File: Python/cryptography/playfair.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        
+    elif ch==2:
+        try:
+            decrypt()
+        except Exception as e:
+            print("Wrong input!")
+    elif ch==3:
+        break;
+    else:
+        print("Choose correct choice")
```
Comment (vatsalkesarwani12) [inline Python/cryptography/playfair.py:120]: add a sample test case in comment

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]

File: Python/cryptography/playfair.py (hunk 5 above)
Comment (sharanya02) [reply Python/cryptography/playfair.py:120]: Done

**Review by vatsalkesarwani12 — APPROVED** [review]

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@sharanya02 please do the following changes:
1. Resolve the conflict
2. I am seeing here a lot of unnecessary changes in readme, please revert these and make only the changes required for this PR.

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@sharanya02 please incorporate the suggested changes.

**Comment by atarax665** [comment]
I am closing this PR due to inactivity from the contributor's side.

**Pull request closed without merging**